src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseLinksControllerTest.java

```diff
@@ -0,0 +1,127 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import uk.gov.hmcts.et.common.model.ccd.CCDRequest;
+import uk.gov.hmcts.et.common.model.ccd.CaseDetails;
+import uk.gov.hmcts.et.common.model.ccd.SubmitEvent;
+import uk.gov.hmcts.ethos.replacement.docmosis.service.CaseLinksEmailService;
+import uk.gov.hmcts.ethos.replacement.docmosis.service.VerifyTokenService;
+import uk.gov.hmcts.ethos.replacement.docmosis.utils.JsonMapper;
+import uk.gov.hmcts.ethos.utils.CCDRequestBuilder;
+import uk.gov.hmcts.ethos.utils.CaseDataBuilder;
+
+import java.util.stream.Stream;
+
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.ENGLANDWALES_CASE_TYPE_ID;
+
+@ExtendWith(SpringExtension.class)
+@WebMvcTest({CaseLinksController.class, JsonMapper.class})
+class CaseLinksControllerTest {
+
+    private static final String AUTH_TOKEN = "Bearer eyJhbGJbpjciOiJIUzI1NiJ9";
+    private static final String CREATE_SUBMITTED_URL = "/caseLinks/create/aboutToSubmit";
+
+    private static final String MAINTAIN_SUBMITTED_URL = "/caseLinks/maintain/aboutToSubmit";
+
+    @MockBean
+    private VerifyTokenService verifyTokenService;
+    @MockBean
+    private CaseLinksEmailService caseLinksEmailService;
+
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private JsonMapper jsonMapper;
+    private CCDRequest ccdRequest;
+
+    @BeforeEach
+    void setUp() throws Exception {
+
+        CaseDetails caseDetails = CaseDataBuilder.builder()
+                .buildAsCaseDetails(ENGLANDWALES_CASE_TYPE_ID);
+
+        caseDetails.getCaseData().setEthosCaseReference("1234");
+
+        ccdRequest = CCDRequestBuilder.builder()
+                .withCaseData(caseDetails.getCaseData())
+                .build();
+
+        SubmitEvent submitEvent = new SubmitEvent();
+        submitEvent.setCaseData(caseDetails.getCaseData());
+
+        when(verifyTokenService.verifyTokenSignature(AUTH_TOKEN)).thenReturn(true);
+    }
+
+    @ParameterizedTest
+    @MethodSource("requests")
```

Comment:
This is cool, makes our controller tests file shorter. I'm not sure where we sit with these tests now though as @furnivall  disagreed with apiTests for these controllers, is it worth testing that methods are called and/or case data state *after* running the callback (obvs controller dependant here)

Reply:
@JackReevies I see your point. For me the main thing this is testing is the request. Should anyone modify the controller(url or body) I'd want these tests to start failing.
For this specific Controller test I don't want to test case data as its not something we are modifying and service returns void